nexus/mgs-updates/src/driver_update.rs

```diff
@@ -212,19 +217,34 @@ pub(crate) async fn apply_update(
     debug!(log, "loaded artifact contents");
 
     // Check the live state first to see if:
-    // - this update has already been completed, or
+    // - this update has already been completed,
+    // - we are waiting for an ongoing update, or
     // - if not, then if our required preconditions are met
     status.update(UpdateAttemptStatus::Precheck);
-    match update_helper.precheck(log, &mut mgs_clients, update).await {
-        Ok(PrecheckStatus::ReadyForUpdate) => (),
-        Ok(PrecheckStatus::UpdateComplete) => {
-            return Ok(UpdateCompletedHow::FoundNoChangesNeeded);
-        }
-        Err(error) => {
-            return Err(ApplyUpdateError::PreconditionFailed(error));
-        }
-    };
+    let before = Instant::now();
+    loop {
+        match update_helper.precheck(log, &mut mgs_clients, update).await {
+            Ok(PrecheckStatus::ReadyForUpdate) => break,
+            Ok(PrecheckStatus::WaitingForOngoingUpdate) => {
+                if before.elapsed() >= progress_timeout {
+                    warn!(
+                        log,
+                        "update takeover: timed out while waiting for ongoing update"
```

Comment:
Do we have a test for this case?

Reply:
Not necessary anymore since we removed this logic